src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAroundCheck.java

```diff
@@ -690,17 +690,18 @@ public void visitToken(DetailAST ast) {
             final String line = getLine(ast.getLineNo() - 1);
             final int before = ast.getColumnNo() - 1;
             final int after = ast.getColumnNo() + ast.getText().length();
+            final int[] codePoints = line.codePoints().toArray();
 
             if (before >= 0) {
-                final char prevChar = line.charAt(before);
+                final char prevChar = Character.toChars(codePoints[before])[0];
```

Comment:
Why can't we use `codePointAt` and `codePoints().count()`. Are they any faster/better than always converting the entire line to code points?

Reply:
> Why can't we use codePointAt and codePoints().count() ?

This would make check logic more complicated - we would need to translate the indexes between chars and codepoints to get the correct index of the character we want to check. Sadly, java string/ character array representation is a bit behind the times, because it still uses surrogate pairs in representing UTF-16 high code points. From `codePointAt(int index)` javadoc: `The index refers to char values (Unicode code units) and ranges from 0 to length()- 1.`

The approach I have used is more similar to the existing logic, and likely easier to maintain in the future, because ` Character.toChars()` gets the entire codepoint (i.e. one emoji or character, as it is seen in the source file, with corresponding index).

Using `codePointAt()` would also likely lead to more complications down the road, since we do not know which member of the surrogate pair we are getting.

> Are they any faster/better than always converting the entire line to code points?

Yes, performance would likely be better if we could only use these methods (especially `count()`, it can do this in parallel, or depending on the type of `IntStream` it may already know the number of code points), but when accounting for the necessary translation, I doubt it.